common/lib/xmodule/xmodule/modulestore/__init__.py

```diff
@@ -409,6 +411,37 @@ class ModuleStoreAssetWriteInterface(ModuleStoreAssetInterface):
     """
     The write operations for assets and asset metadata
     """
+    def _save_assets_by_type(self, course_key, asset_metadata_list, course_assets, user_id, import_only):
+        """
+        Common private method that saves/updates asset metadata items in the internal modulestore
+        structure used to store asset metadata items.
+        """
+        assets_by_type = {}
```

Comment:
fyi: I believe you could remove L428-429 if you do this as
```
  assets_by_type = defaultdict(lambda: SortedListWithKey(course_assets.get(asset_type, []), key=itemgetter('filename')))
```
but python variable scoping confuses me enough to not be certain